server/pages/index.js

```diff
@@ -342,13 +342,84 @@ function setUpLoggedInRoute( req, res, next ) {
 	}
 }
 
-function setUpRoute( req, res, next ) {
-	if ( req.cookies.wordpress_logged_in ) {
-		// the user is probably logged in
-		setUpLoggedInRoute( req, res, next );
-	} else {
-		setUpLoggedOutRoute( req, res, next );
+/**
+ * Sets up a Content Security Policy header
+ * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP
+ * @param {Object} req Express request object
+ * @param {Object} res Express response object
+ * @param {Function} next a callback to call when done
+ */
+function setUpCSP( req, res, next ) {
+	const originalUrlPathname = req.originalUrl.split( '?' )[ 0 ];
```

Comment:
this is a reasonable compromise, though even more since we're on server code it might make sense to use `url` to actually parse it.

my primary concern was things that this takes care of, like `https://wordpress.com/some/path?redirect_to=/log-in/me` which I'm pretty sure wasn't the intended match of that original RegExp :+1:

Reply:
Yeah, it was a good catch 👍